homeassistant/components/switch/mochad.py

```diff
@@ -50,24 +50,46 @@ def __init__(self, hass, ctrl, dev):
         self._comm_type = dev.get(mochad.CONF_COMM_TYPE, 'pl')
         self.device = device.Device(ctrl, self._address,
                                     comm_type=self._comm_type)
-        self._state = self._get_device_status()
```

Comment:
Why no longer fetch the state on startup?

Reply:
The original development must have been done with cm15a (has both pl and rf). I'm using cm19a (has only rf), which cannot get the status for power line switches. It delays the startup for ca 50 seconds on 10 devices.

Reply:
Ok. fixed in commit 3. e88301b

Reply:
I've made the old and new line conditional. See commit 5. In case of comm_type == 'pl' the status will be retrieved as previously. Both cm15a and cm19a should be fine now.